// jest-mock-axios/dist/cancel/CancelToken.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * Based of https://github.com/axios/axios
 * Copyright (c) 2014-present Matt Zabriskie
 */
var CancelToken =
/*#__PURE__*/
function () {
  _createClass(CancelToken, null, [{
    key: "source",
    value: function source() {
      var cancel;
      var token = new CancelToken(function (c) {
        return cancel = c;
      });
      return {
        cancel: cancel,
        token: token
      };
    }
  }]);

  function CancelToken(executor) {
    _classCallCheck(this, CancelToken);

    executor(function cancel(message) {
      return;
    });
  }

  _createClass(CancelToken, [{
    key: "throwIfRequested",
    value: function throwIfRequested() {
      return;
    }
  }]);

  return CancelToken;
}();

exports["default"] = CancelToken;
